src/main/kotlin/dev/supergrecko/kllvm/core/typedefs/Module.kt

```diff
@@ -87,6 +91,30 @@ public class Module internal constructor() : AutoCloseable,
         LLVM.LLVMDisposeModule(ref)
     }
 
+    //region Analysis
+    /**
+     * Verifies that the module structure is valid
+     *
+     * This function returns true if the module is valid as opposed to the
+     * LLVM implementation which would return 0 if the module is valid.
+     *
+     * TODO: Find a nice way to return the string which the LLVM method returns
+     *   Because of this. When calling this with PrintMessage or ReturnStatus
+     *   the underlying bytes in the ptr are really strange (see #67)
+     */
+    public fun verify(action: VerifierFailureAction): Boolean {
+        val ptr = BytePointer(ByteBuffer.allocate(100))
```

Comment:
I used a size of 100 here for testing because I don't know how to get the proper size.